kvbc/tools/db_editor/include/kv_blockchain_db_editor.hpp

```diff
@@ -0,0 +1,486 @@
+      throw std::invalid_argument{"Missing BLOCK-ID argument"};
+    }
+    const auto raw_block = adapter.getRawBlock(toBlockId(args.values.front()));
+    return toJson("rawBlock",
+                  raw_block ? concordUtils::vectorToHex(categorization::RawBlock::serialize(*raw_block)) : "n/a"s);
+  }
+};
+
+struct GetRawBlockRange {
+  bool read_only = true;
+  std::string description() const {
+    return "getRawBlockRange BLOCK-ID-START BLOCK-ID-END\n"
+           "  Returns a list of serialized raw blocks (encoded in hex) in the [BLOCK-ID-START, BLOCK-ID-END) range.";
+  }
+
+  std::string execute(const KeyValueBlockchain &adapter, const CommandArguments &args) const {
+    if (args.values.size() < 2) {
+      throw std::invalid_argument{"Missing or invalid block range"};
+    }
+    const auto end = toBlockId(args.values[1]);
+    if (end == 0) {
+      throw std::invalid_argument{"Invalid BLOCK-ID-END value"};
+    }
+    const auto first = toBlockId(args.values[0]);
+    const auto last = std::min(end - 1, getLatestBlockId(adapter));
+    if (first > last) {
+      throw std::invalid_argument{"Invalid block range"};
+    }
+    auto raw_blocks = std::vector<std::pair<std::string, std::string>>{};
+    for (auto i = first; i <= last; ++i) {
+      const auto raw_block = adapter.getRawBlock(i);
+      if (!raw_block) {
+        throw NotFoundException{"Couldn't find a block by ID = " + std::to_string(i)};
+      }
+      raw_blocks.emplace_back("rawBlock" + std::to_string(i),
+                              concordUtils::vectorToHex(categorization::RawBlock::serialize(*raw_block)));
+    }
+    return toJson(raw_blocks);
+  }
+};
+
+struct GetBlockInfo {
+  bool read_only = true;
+  std::string description() const {
+    return "getBlockInfo BLOCK-ID\n"
+           "  Returns information about the requested block (excluding its key-values).";
+  }
+
+  std::string execute(const KeyValueBlockchain &adapter, const CommandArguments &args) const {
+    if (args.values.empty()) {
+      throw std::invalid_argument{"Missing BLOCK-ID argument"};
+    }
+    auto blockId = toBlockId(args.values.front());
+    const auto parent_digest = adapter.parentDigest(blockId);
+    const auto updates = adapter.getBlockUpdates(blockId);
+    if (!updates) {
+      throw NotFoundException{"Couldn't find a block by ID = " + std::to_string(blockId)};
+    }
+    auto catUpdates = updates->categoryUpdates();
+    size_t keyValueCount = 0;
+    for (auto const &catkv : catUpdates.kv) {
+      keyValueCount += std::visit([](auto &&arg) { return arg.kv.size(); }, catkv.second);
+    }
+
+    return toJson(std::map<std::string, std::string>{
+        std::make_pair(
+            "parentBlockDigest",
+            parent_digest ? concordUtils::bufferToHex(parent_digest->data(), parent_digest->size()) : "n/a"s),
+        std::make_pair("keyValueCount", std::to_string(keyValueCount)),
+        std::make_pair("categoriesCount", std::to_string(catUpdates.kv.size()))});
```

Comment:
I think what might be useful is to report key counts per category. Don't think it is critical for this PR, but just something I think people would want.